caravel/views.py

```diff
@@ -1964,6 +1964,38 @@ def cached_key(self, key):
         return "nope"
 
     @has_access_api
+    @expose("/results/<key>/")
+    @log_this
+    def results(self, key):
```

Comment:
if we'll have key as a query_name + timestamp we would be able to tell the user if the result set was expired